README.md

```diff
@@ -1,17 +1,301 @@
+  - [Example: Rebasing an App Image](#example-rebasing-an-app-image)
+  - [Rebasing Explained](#rebasing-explained)
+- [Working with Builders Using `create-builder`](#working-with-builders-using-create-builder)
+  - [Example: Creating a Builder from Buildpacks](#example-creating-a-builder-from-buildpacks)
+  - [Builders Explained](#builders-explained)
+- [Managing Stacks](#managing-stacks)
+  - [Example: Adding a Stack](#example-adding-a-stack)
+  - [Example: Updating a Stack](#example-updating-a-stack)
+  - [Example: Deleting a Stack](#example-deleting-a-stack)
+  - [Example: Setting the Default Stack](#example-setting-the-default-stack)
+  - [Listing Stacks](#listing-stacks)
+- [Resources](#resources)
+- [Development](#development)
 
-## Example Usage
+----
 
+## Building App Images Using `build`
+
+`pack build` enables app developers to create runnable app images from source code using buildpacks.
+
+```bash
+$ pack build <image-name>
 ```
-./pack build packs/myimage:mytag --path ./myapp
+
+### Example: Building Using the Default Builder Image
+
+In the following example, an app image is created from Node source code.
+
+```bash
+$ cd /path/to/node/app
+$ pack build my-app:my-tag
+
+# ... Detect, analyze and build output
+
+Successfully built 2452b4b1fce1
+Successfully tagged my-app:my-tag
 ```
 
-The above will create images on your local daemon. If you wish to export images directly to a docker registry, use the `--publish` flag.
+In this case, the default [builder](#working-with-builders-using-create-builder) is used, and an appropriate buildpack
+is automatically selected from the builder based on the app source code. To understand more about what builders are and
+how to create or use them, see the
+[Working with Builders Using `create-builder`](#working-with-builders-using-create-builder) section.
 
+To publish the produced image to an image registry, include the `--publish` flag (this is most useful
+for private registries):
+
+```bash
+$ pack build private-registry.example.com/my-app:my-tag --publish
 ```
-./pack build packs/myimage:mytag --path ./myapp --publish
+
+### Example: Building Using a Specified Buildpack
+
+In the following example, an app image is created from Node source code, using a buildpack chosen by the user.
+
+```bash
+$ cd /path/to/node/app
+$ pack build my-app:my-tag --buildpack path/to/some/buildpack
+
+# ...
+*** DETECTING WITH MANUALLY-PROVIDED GROUP:
+2018/10/29 18:31:05 Group: Name Of Some Buildpack: pass
+# ...
+
+Successfully built 2452b4b1fce1
+Successfully tagged my-app:my-tag
 ```
+
+The message `DETECTING WITH MANUALLY-PROVIDED GROUP` indicates that the buildpack was chosen by the user, rather than
+by the automated detection process.
+
+The `--buildpack` parameter can be
+- a path to a directory
+- a path to a `.tgz` file
+- a URL to a `.tgz` file, or
+- the ID of a buildpack located in a builder
+
+### Building Explained
+
+![](docs/build-diagram.png)
```

Comment:
Add alt-text for accessibility.

Reply:
Also, might consider SVGs instead of raster images.

Reply:
@sclevine 
Re: alt-text - will do.
Re: SVGs, I've had trouble in the past putting SVGs in readmes on github, but I'll try it out